src/tools/pkcs11-tool.c

```diff
@@ -4822,6 +4808,10 @@ static CK_RV write_object(CK_SESSION_HANDLE session)
 				type = CKK_GENERIC_SECRET;
 			else if (strncasecmp(opt_key_type, "HKDF:", strlen("HKDF:")) == 0)
 				type = CKK_HKDF;
+			else if (strncasecmp(opt_key_type, "CHACHA20", strlen("CHACHA20")) == 0)
+				type = CKK_CHACHA20;
+			else if (strncasecmp(opt_key_type, "POLY1305", strlen("POLY1305")) == 0)
+				type = CKK_POLY1305;
```

Comment:
Please add documentation of the new values here:
https://github.com/OpenSC/OpenSC/blob/9b0313d3caf154b1506e90a03d92cd34d176b749/doc/tools/pkcs11-tool.1.xml#L146-L151